src/python_framework_v02/troute/network/musking/mc_reach.pyx

```diff
@@ -0,0 +1,95 @@
+cimport numpy as np
+import numpy as np
+from libc.stdlib cimport malloc, free
+from troute.network.reach cimport compute_type, _Reach
+
+cdef extern from "mc_reach_structs.c":
+  void init_mc_reach(_Reach* reach, int num_segments)
+  void free_mc_reach(_Reach* reach)
+  void set_mc_segment(_Reach* reach, int index, long id,
+      float dt, float dx, float bw, float tw, float twcc,
+      float n, float ncc, float cs, float s0,
+      float qdp, float velp, float depthp)
+  _MC_Segment get_mc_segment(_Reach* reach, int index) nogil
+
+cdef class MC_Segment(Segment):
```

Comment:
I don't think this is required/used.

Reply:
Not so true.  MC_Segment is currently used to build up the reach from a list of segments.  This could be easily refactored to use the same pure C lib struct...